portable/ThirdParty/GCC/Posix/port.c

```diff
@@ -396,6 +452,12 @@ void prvSetupTimerInterrupt( void )
 
 static void vPortSystemTickHandler( int sig )
 {
+    if( prvIsFreeRTOSThread( pthread_self() ) == pdFALSE )
+    {
+        fprintf( stderr, "vPortSystemTickHandler called from non-FreeRTOS thread\n" );
+        return;
```

Comment:
Although personally I think this early return makes the code more readable, could you instead just use an if else? This is purely to make static analyzers happy as MISRA complains about multiple returns.

Reply:
Ok I can do that!

Reply:
Done